manager/integration/tests/test_ha.py

```diff
@@ -859,4 +860,75 @@ def test_single_replica_failed_during_engine_start():
     11. Check if the volume still works fine by
         r/w data and creating/removing snapshots.
     """
-    pass
+    pv_name = "pv-" + volume_name
+    pvc_name = "pvc-" + volume_name
+    pod_name = "pod-" + volume_name
+
+    volume = create_and_check_volume(client, volume_name)
+    create_pv_for_volume(client, core_api, volume, pv_name)
+    create_pvc_for_volume(client, core_api, volume, pvc_name)
+
+    pod['metadata']['name'] = pod_name
+    pod['spec']['volumes'] = [{
+        'name': pod['spec']['containers'][0]['volumeMounts'][0]['name'],
+        'persistentVolumeClaim': {
+            'claimName': pvc_name,
+        },
+    }]
+
+    create_and_wait_pod(core_api, pod)
+    wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+    test_data1 = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_volume_data(core_api, pod_name, test_data1)
+    volume.snapshotCreate()
+    test_data2 = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_volume_data(core_api, pod_name, test_data2)
+    snap2 = volume.snapshotCreate()
+    test_data3 = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_volume_data(core_api, pod_name, test_data3)
+    volume.snapshotCreate()
+
+    volume = client.by_id_volume(volume_name)
+    host_id = get_self_host_id()
+
+    for replica in volume.replicas:
+        if replica.hostId == host_id:
+            break
+
+    replica_data_path = replica.dataPath
+    replica_name = replica.name
+    snap2_meta_file = replica_data_path + \
+        "/volume-snap-" + \
+        snap2.name + ".img.meta"
+
+    command = ["dd", "if=/dev/zero",
+               "of="+snap2_meta_file,
+               "count=" + str(VOLUME_RWTEST_SIZE)]
+    subprocess.check_call(command)
+
+    delete_and_wait_pod(core_api, pod_name)
+    wait_for_volume_detached(client, volume_name)
+
+    create_and_wait_pod(core_api, pod)
+    wait_for_volume_degraded(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+    for repl in volume.replicas:
+        if repl.name == replica_name:
+            break
+
+    assert repl.running is False
+    assert repl.failedAt != ''
+
+    wait_for_volume_healthy(client, volume_name)
+    vol_data = read_volume_data(core_api, pod_name)
```

Comment:
I prefer to use these functions to write and check data:
```
write_pod_volume_random_data(core_api, pod_name,
                                 data_path, DATA_SIZE_IN_MB_1)
get_pod_data_md5sum(core_api, pod_name, data_path)
```

Then we can check all data written to the volume rather than the last written data.